src/robotFactory.js

```diff
@@ -1,10 +1,77 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    if (coords) {
+      if (!coords.x) {
+        coords.x = 0;
+      }
 
-class DeliveryDrone {}
+      if (!coords.y) {
+        coords.y = 0;
+      }
+      this.coords = coords;
```

Comment:
```suggestion
      this.coords = { ...coords };
```

let's not mutate the argument, in case it is used somewhere else